flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/TransformFilterProcessor.java

```diff
@@ -135,6 +159,19 @@ private Object[] generateParams(BinaryRecordData record, long epochTime, String
                     params.add(opType);
                     continue;
             }
+
+            boolean foundInMeta = false;
+            for (SupportedMetadataColumn supportedMetadataColumn : supportedMetadataColumns) {
+                if (supportedMetadataColumn.getName().equals(columnName)) {
+                    params.add(supportedMetadataColumn.read(meta));
+                    foundInMeta = true;
+                    break;
+                }
+            }
+            if (foundInMeta) {
+                continue;
+            }
```

Comment:
Is it possible to let `supportedMetadataColumns()` return a `Map<String, SupportedMetadataColumn>`? This code would be executed for each stream data record and this array traversing seems a little inefficient.

Reply:
`Map` is not `Serializable` for `TransformRule`. I could build a map for this class.

Reply:
`UnmodifiableMap` could be an option. Of course building one dynamically is fine, too.